packages/core-modules/test/contracts/modules/FeatureFlagModule.test.js

```diff
@@ -0,0 +1,137 @@
+const { ethers } = hre;
+const { ethers: Ethers } = require('ethers');
+const assertBn = require('@synthetixio/core-utils/utils/assertions/assert-bignumber');
+const { default: assertEvent } = require('@synthetixio/core-utils/utils/assertions/assert-event');
+const { default: assertRevert } = require('@synthetixio/core-utils/utils/assertions/assert-revert');
+const { bootstrap } = require('@synthetixio/hardhat-router/dist/utils/tests');
+const initializer = require('@synthetixio/core-modules/test/helpers/initializer');
+
+describe('FeatureFlagModule', () => {
+  const { proxyAddress } = bootstrap(initializer, {
+    modules: '.*(FeatureFlagModule|SampleFeatureFlagModule|Owner|Upgrade).*',
+  });
+
+  let FeatureFlagModule, SampleFeatureFlagModule;
+  let owner, permissionedUser, user;
+
+  const FEATURE_FLAG_NAME = Ethers.utils.formatBytes32String('SAMPLE_FEATURE');
+
+  before('identify signers', async () => {
+    [owner, permissionedUser, user] = await ethers.getSigners();
+  });
+
+  before('identify modules', async () => {
+    FeatureFlagModule = await ethers.getContractAt('FeatureFlagModule', proxyAddress());
+    SampleFeatureFlagModule = await ethers.getContractAt('SampleFeatureFlagModule', proxyAddress());
+  });
+
+  it('does not allow non-owners to set feature flags', async () => {
+    await assertRevert(
+      FeatureFlagModule.connect(user).setFeatureFlag(FEATURE_FLAG_NAME, true),
+      'Unauthorized'
+    );
+  });
+
+  let setupTx;
+  before('setup feature flag', async () => {
```

Comment:
this before (afaik) runs before *any* of the `it` clauses in this describe, so its confusing to put it in the middle here

Reply:
please let me know if I am wrong about taht

Reply:
yep you right; moving it to the top